src/renderer.rs

```diff
@@ -600,6 +600,17 @@ pub fn pick(
     )
 }
 
+/// Result from an intersection test
+#[derive(Debug, Clone, Copy)]
+pub struct IntersectionResult {
+    /// The position of the intersection
+    pub position: Vec3,
+    /// The index of the intersected geometry in the list of geometries
+    pub geometry_id: u32,
+    /// The index of the intersected instance in the list of instances or 0 if the intersection did not hit an instanced geometry
+    pub instance_id: u32,
+}
```

Comment:
I recently noticed other 3D rendering libraries that provide this functionality also include the face index in [their return result](https://developer.apple.com/documentation/scenekit/scnhittestresult), which would take up the last alpha color slot. I believe this can be obtained from `gl_PrimitiveID` in the `IntersectionMaterial` fragment shader. This isn't needed for my use case, but could be beneficial in this hypothetical use case: a button panel, where all the buttons are part of the same model; it is known which triangles make up the buttons, so the face index is used to identify which button was clicked.

Reply:
That's a good point! No reason not to add that to the intersection result as well. Fixed in [e390c6a](https://github.com/asny/three-d/pull/494/commits/e390c6af4e3ed7a80778193aaa8c0cdbd88d6e9d)

Reply:
Turns out `gl_PrimitiveID` is not supported on web, so I'm removing it again.